### PR TITLE
Refactor CheckboxList component

### PR DESCRIPTION
**Pull request by sBouzols**
Make it a complete controlled component and remove complex useEffects which were there to synchronize the internal `checked` state and an external one updated with the `onChecked` callback.
Use a Set to avoid duplicates at insert, not inside this component.

### CODE REVIEW

File: src/components/dialogs/contingency-list-selector.jsx [hunk 1]
```diff
@@ -146,30 +141,25 @@ const ContingencyListSelector = (props) => {
     };
 
     const removeFromFavorite = (toRemove) => {
-        const toDelete = new Set(toRemove);
```
Comment (achour94) [inline src/components/dialogs/contingency-list-selector.jsx:149]: The Set performs existence checks more efficiently than Array.includes, i think this is the purpose of doing this

File: src/components/dialogs/contingency-list-selector.jsx [hunk 2]
```diff
@@ -146,30 +141,25 @@ const ContingencyListSelector = (props) => {
...
         saveFavorite(
             contingencyList
                 .map((e) => e.id)
-                .filter((item) => !toDelete.has(item))
+                .filter((item) => !toRemove.includes(item))
```
Comment (achour94) [inline src/components/dialogs/contingency-list-selector.jsx:147]: why you dont filter directly like this contingencyList.filter((item) => !toRemoveSet.includes(item.id))

File: src/components/dialogs/contingency-list-selector.jsx [hunk 3]
```diff
@@ -146,30 +141,25 @@ const ContingencyListSelector = (props) => {
...
         );
-        const newChecked = checkedContingencyListUuids.filter(
-            (item) => !toDelete.has(item)
+
+        setCheckedContingencyList((oldCheked) =>
+            oldCheked.filter((item) => !toRemove.includes(item))
```
Comment (achour94) [inline src/components/dialogs/contingency-list-selector.jsx:151]: you are checking if an array of ids includes an object (item)

File: src/components/dialogs/contingency-list-selector.jsx [hunk 4]
```diff
@@ -146,30 +141,25 @@ const ContingencyListSelector = (props) => {
     };
 
     const removeFromFavorite = (toRemove) => {
```
Comment (achour94) [inline src/components/dialogs/contingency-list-selector.jsx:143]: this function receive sometimes an array of ids, and another time an array of objects, for concistency with addFavorites use the same pattern

File: src/components/dialogs/contingency-list-selector.jsx (hunk 1 above)
Comment (sBouzols) [reply src/components/dialogs/contingency-list-selector.jsx:149]: Those are really litlle arrays, I wanted to keep it simple but ths is right. will revert

File: src/components/utils/checkbox-list.jsx [hunk 5]
```diff
@@ -5,74 +5,41 @@
...
     values,
     onChecked,
-    initialSelection,
-    itemComparator = (a, b) => a.id === b.id,
+    checkedValues,
+    itemComparator = defaultComparator,
     ...props
 }) => {
-    const [checked, setChecked] = useState(new Set(initialSelection));
+    const valuesRef = useRef();
```
Comment (achour94) [inline src/components/utils/checkbox-list.jsx:24]: i don't see where this variable are used !?

File: src/components/utils/checkbox-list.jsx [hunk 6]
```diff
@@ -5,74 +5,41 @@
...
-        [checked, onChecked, itemComparator]
-    );
-
-    useEffect(() => onChecked && onChecked(checked), [checked, onChecked]);
-
-    const isCheckboxInCheckedSet = (checkedSet, checkBoxToCheck) => {
-        return Array.from(checkedSet).some((element) =>
-            itemComparator(element, checkBoxToCheck)
-        );
+        onChecked([...newCheckedValues]);
```
Comment (achour94) [inline src/components/utils/checkbox-list.jsx:42]: newCheckedValues is already a new array , no need to spread it again

File: src/components/utils/checkbox-list.jsx (hunk 5 above)
Comment (sBouzols) [reply src/components/utils/checkbox-list.jsx:24]: arf forget to clean this

File: src/components/dialogs/contingency-list-selector.jsx (hunk 1 above)
Comment (sBouzols) [reply src/components/dialogs/contingency-list-selector.jsx:149]: fixed

File: src/components/dialogs/contingency-list-selector.jsx (hunk 3 above)
Comment (sBouzols) [reply src/components/dialogs/contingency-list-selector.jsx]: Should be Array of Objects as arguments for `addFavorites` and `removeFromFavorites` now

File: src/components/dialogs/contingency-list-selector.jsx (hunk 4 above)
Comment (sBouzols) [reply src/components/dialogs/contingency-list-selector.jsx]: Should be Array of Objects as arguments for addFavorites and removeFromFavorites now

File: src/components/dialogs/contingency-list-selector.jsx (hunk 2 above)
Comment (sBouzols) [reply src/components/dialogs/contingency-list-selector.jsx]: saveFavorites expect only some ids, then we need to filter and map.

**Review by achour94 — APPROVED** [review]

**Review by thangqp — APPROVED** [review]

**Pull request merged**